language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -93,11 +93,11 @@ optionsParser = Options
         <> metavar "DIR"
         <> help "Output directory for the generated packages"
         )
-    <*> optional (strOption
+    <*> strOption
         (  short 'p'
         <> metavar "PACKAGE-JSON"
         <> help "Path to an existing 'package.json' to update"
```

Comment:
The docs are not right, you will never update this file now. This raises the question whether this is really the behavior we want. We definitely do not want to overwrite a file if the user has specified nothing. However, if I specify the flag explicitly it does seem weird that it’s ignored if the file already exists.

How about the following:
1. Make the flag optional. If it is not specified, create `package.json` if it does not already exist.
2. If the flag is passed explicitly keep the current behavior of overwriting the config.

Reply:
> The docs are not right, you will never update this file now

I'm not seeing that?

I do agree with the rest.

Reply:
> Path to an existing package.json to update

You check if the file exists and ignore it in that case. So you might create a new file but you will never update it.

Reply:
Oh these docs : "Path to an existing 'package.json' to update""?

Reply:
How about this:
```
    <*> strOption
        (  short 'p'
        <> metavar "PACKAGE-JSON"
        <> value "package.json"
        <> help "Path to a 'package.json' to create/update"
        )
```
and otherwise the logic is as it is now - creating the file if doesn't exist and regardless, updating the file when packages have been generated.

Reply:
So to recap with this approach:
  - if there is no flag, if "package.json" doesn't exist it will be created and later in the program updated;
  - if there is a flag and that package.json doesn't exist it will be created and later in the program updated;
 - if there is a flag and the package.json exists, it will not be recreated but will later in the program updated.

Reply:
> You check if the file exists and ignore it in that case. So you might create a new file but you will never update it.

I think that's where the confusion lies. The file `optInputPackage.json` is read from and written to in `setupWorkspace`.

Reply:
Oh I see, I didn’t realize that.

Reply:
The new phrasing moves the creation into `setupWorkspace` itself so hopefully that won't confuse anyone again.